client/src/components/EmojiPicker.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+import React from 'react';
+import ReactEmoji from "react-emoji";
+import { filterEmojis } from "../utils/emojis";
+import 'styles/EmojiPicker.scss';
+
+const emojify = (emojis, highlightedIndex) => {
+    const emojiOpts = {
+      emojiType: 'emojione'
+    };
+    let counter = 0;
+    return emojis.map(emoji => {
+            let elementClass = '';
+            const element = emoji.element;// ReactEmoji.emojify(emoji, emojiOpts)[0];
+            if (counter === highlightedIndex)
+                elementClass = 'selected';
+            return (<li key={counter++}
+                        className={elementClass}>
+                        {element}
+                    </li>);
+        }
+    );
+};
+
+const EmojiList = ({
+    emojis,
+    highlightedIndex
+}) => <ul> { emojify(emojis, highlightedIndex) } </ul>
+
+class EmojiPicker extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            emojis: [],
+            highlightedIndex: 0
+        }
+    }
+
```

Comment:
Instead of "manually" handling the emoji filtering, we can hook up to React component's lifecycle methods `componentDidMount` (called after the component was "added" to the DOM the first time) and `componentWillReceiveProps` (called everytime new properties (this.props) are passed to the component). By using these two, we can get the `lastWord` from `SendMessage` as a property and react to changes in it. 

Let's call the passed property `filterText`. What we want, when the passed text changes, is to find the emojis from the `emojiList` we created above. We can create a simple filtering function (put this somewhere at the top of the file):
```javascript
const filterEmojis = function(emojiList, word, amount) {
  return Object.values(emojiList)
    .filter((e) => e.shortname.indexOf(word) > -1 || e.aliases.indexOf(word) > -1 || word === '')
    .slice(0, amount)
}
```
Notice how we can check for matches from multiple fields in the emojilist element properties. We could extend this to search from the `keywords` field, too. Also notice that we return the emoji is the `word` is empty so that we can display all emojis in case the user only has written ':'

Then we can re-filter the list of emojis when the `filterText` property changes.
```javascript
  componentDidMount() {
    const emojis = filterEmojis(emojiList, this.props.filterText, 100);
    this.setState({ emojis: emojis });
  }

  componentWillReceiveProps(nextProps) {
    const emojis = filterEmojis(emojiList, nextProps.filterText, 100);
    this.setState({ emojis: emojis });
    // call the 'onClose' callback when we have no matches or if the filter is empty
    if(nextProps.filterText === '' || emojis.length === 0)
      this.props.onClose();
  }
```